cluster-service/generate_helm_set_flags.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash 
+
+# The 'roleName' field in the ../config/config.yaml file is a comma separated list of role names.
+# This script separates them and generates `--set` flags for the `helm deploy` command.
+# Please note that since here, the role names are being separated by commas, the role name itself
+# cannot have a comma in it's name.
+
+genereate_set_flags() {
+    HELM_SETS=()
+    IFS=',' read -a ROLE_NAMES <<< "${2}" 
+    for ((i=0; i<${#ROLE_NAMES[@]}; i++)); do
+        ROLE_NAME="${ROLE_NAMES[$i]}"
+        HELM_SETS+=(--set "azureOperatorsMI.${1}.roleDefinitions[$i].name=${ROLE_NAME}") 
+        ID=$(az role definition list --name "${ROLE_NAME}" --query "[].name" -o tsv)
+        HELM_SETS+=(--set "azureOperatorsMI.${1}.roleDefinitions[$i].id=${ID}")
+    done
+    printf "%q " "${HELM_SETS[@]}"
+}
+
+OP_HELM_SET_FLAGS=()
```

Comment:
I would do all of this only when explicitly called

Reply:
Sorry, can you please elaborate?

We are already doing this only when 2 of the make targets(`deploy` and `local-azure-operators-managed-identities-config`) are executed and where this is required.

Reply:
The idea is for the file to only define functions. Those functions would be available to the consumers of this file. Including/sourcing this file would not generate any side / implicit effects, which can be harder to follow and reason about when issues arise.

Reply:
As discussed in our meet. We are going to keep the function as is.